lib/vector-core/src/event/value.rs

```diff
@@ -22,6 +23,41 @@ pub enum Value {
     Null,
 }
 
+impl Eq for Value {}
+
+impl Hash for Value {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self {
+            Value::Array(v) => {
+                v.hash(state);
+            }
+            Value::Boolean(v) => {
+                v.hash(state);
+            }
+            Value::Bytes(v) => {
+                v.hash(state);
+            }
+            Value::Float(v) => {
+                if v.is_finite() {
+                    format!("{}", v).hash(state);
+                } else {
+                    "NaN".hash(state)
+                }
+            }
+            Value::Integer(v) => {
+                v.hash(state);
+            }
+            Value::Map(v) => {
+                v.hash(state);
+            }
+            Value::Null => {}
```

Comment:
We should ensure that this hashes to _something_. The `core::mem::discriminant().hash()` that I mentioned above should guarantee that.